src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -55,8 +63,15 @@ static enum {
         ACTION_WALL
 } arg_action = ACTION_QUERY;
 
+static volatile sig_atomic_t sigchild;
+
+static void chld_handler(int sig) {
+        ++sigchild;
+}
+
```

Comment:
Sorry I missed this earlier, but your "sigchild" counter is used nowhere... It can really go. And if all you need is a NOP signal handler, then please use `nop_signal_handler()` from signal-util.h